web-server/sgx-wallet-test/enclave/src/ported/proptest_crypto.rs

```diff
@@ -0,0 +1,37 @@
+use std::boxed::Box;
+use std::format;
+
+use proptest::prelude::*;
+use secrecy::{ExposeSecret, Secret};
+use sgx_wallet_impl::ported::crypto::SodaBoxCrypto;
+
+/// Encrypt and decrypt a message using [`SodaBoxCrypto`].
+pub(crate) fn prop_soda_box_roundtrips() {
+    proptest!(|(sender_seed: [u8; 32], receiver_seed: [u8; 32], plaintext: Box<[u8]>)| {
+        prop_assume!(sender_seed != receiver_seed);
+        prop_soda_box_roundtrips_impl(sender_seed, receiver_seed, plaintext);
+    });
+}
+
+fn prop_soda_box_roundtrips_impl(
+    sender_seed: [u8; 32],
+    receiver_seed: [u8; 32],
+    plaintext: Box<[u8]>,
+) {
+    let mut sender = SodaBoxCrypto::from_seed(sender_seed);
+    let receiver = SodaBoxCrypto::from_seed(receiver_seed);
+    assert_ne!(sender.get_pubkey(), receiver.get_pubkey());
+
+    let encrypted = sender
+        .encrypt_message(&Secret::new(plaintext.clone()), &receiver.get_pubkey())
+        .unwrap();
+    let decrypted = receiver
+        .decrypt_message(
+            &encrypted.ciphertext,
+            &sender.get_pubkey(),
+            &encrypted.nonce,
+        )
+        .unwrap();
+
+    assert_eq!(decrypted.expose_secret(), &plaintext);
```

Comment:
For assertions related to the test we might want to use the proptest assert macros:
https://altsysrq.github.io/rustdoc/proptest/latest/proptest/macro.prop_assert.html
https://altsysrq.github.io/rustdoc/proptest/latest/proptest/macro.prop_assert_eq.html

It does require the functions to return a `Result<_, TestCaseError>` instead since it does not panic, but I think that should be fine.

Reply:
Thanks for spotting! We can't use the closure style directly, but we can use the wrapper: fc798e6